buildSrc/src/main/kotlin/org/gradle/plugins/buildtypes/BuildTypesPlugin.kt

```diff
@@ -0,0 +1,62 @@
+package org.gradle.plugins.buildtypes
+
+import org.gradle.api.GradleException
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.kotlin.dsl.extra
+
+class BuildTypesPlugin : Plugin<Project> {
+
+    override fun apply(project: Project): Unit = project.run {
+        val buildTypesContainer = project.container(BuildType::class.java)
```

Comment:
The Gradle Kotlin API should probably expose a version of the API that takes `Klass` or a reified method:
```
val buildTypesContainer = project.container<BuildType>()
```